scripts/generators/match.js

```diff
@@ -0,0 +1,46 @@
+// @ts-check
+
+/**
+ * Generate an articles index matching articles ID with their agreement ID.
+ */
+
+import fs from "fs";
+import log from "npmlog";
+import path from "path";
+import unistUtilFlatFilter from "unist-util-flat-filter";
+
+import { getAgreement, getAgreements } from "../..";
+
+log.enableColor();
+
+log.info("match()", `Indexing articles…`);
+const agreementsIndex = getAgreements();
+const articlesIndex = agreementsIndex.reduce((prevArticlesIndex, { id: agreementId }) => {
```

Comment:
In fact, I answered too quickly here. The agreement index is not a tree, it's a simple collection of ids . I also generally prefer to avoid `reduce()` for the sake of readability and performances but I didn't find anything better. I don't see how I can use flatmap to achieve the same result without overloading everything. Since it's just a script and I don't have so much time left, I will skip that ;)